include/ansi_x931_aes128.c

```diff
@@ -0,0 +1,91 @@
+#include "ansi_x931_aes128.h"
+#include "tiny-AES128-C/aes.h"
+
+#include <string.h>
+#include <stdlib.h>
+
+// TODO: CROSS PLATFORM?
+
+// Some helpers for the prng
+void cgc_gen_block(cgc_prng*);
+void cgc_xor(const uint8_t*, const uint8_t*, uint8_t*);
+
+/**
+ * Initializes a PRNG from a given seed
+ * @param  seed Initial state of the PRNG
+ * @return      Generated PRNG
+ */
+cgc_prng cgc_init_prng(const uint8_t *seed) {
+    cgc_prng prng;
+    memcpy(&prng.state, seed, sizeof(cgc_aes_state));
```

Comment:
There's a type mismatch here. prng.state is a cgc_aes_state, which happens to be a uint8_t[BLOCK_SIZE * 3]. 

Can you use cgc_aes_state everywhere? The caller for this defines it as uint8_t arrays, but they should match.